modules/src/mock/context.rs

```diff
@@ -446,26 +470,30 @@ impl MockContext {
         let new_block = HostBlock::generate_block(
             self.host_chain_id.clone(),
             self.host_chain_type,
-            self.latest_height.increment().revision_height,
+            self.pending_block.height().increment().revision_height,
+            self.pending_block.timestamp().add(self.block_time).unwrap(),
         );
 
+        let old_pending_block = replace(&mut self.pending_block, new_block);
```

Comment:
It does not seem like we need the `pending_block` to have a cached next block. The only place I see it is used is in `host_timestamp`, which should be able to mock the time in other way. Here we can just generate the next block on the fly using the same time offset as used by `host_timestamp`.

Reply:
Done! 👍 I also noticed that `latest_height` wasn't required since we can get the latest height from the history, and we expect the history to be non-empty -> https://github.com/informalsystems/ibc-rs/pull/1771/files#diff-a0183026217183d9147ad2c8fc0827f60c1e4354e9a2bed9da05f5f20fc501b6L842